ParseStarterProject/src/com/parse/starter/LoginActivity.java

```diff
@@ -40,7 +40,7 @@ private void login() {
         String username = usernameEditText.getText().toString().trim();
         String password = passwordEditText.getText().toString().trim();
 
-        if(username.length() != 0 && password.length() != 0) {
+        if(username.length() != 0 && password.length() != 0) {  //requires user and pass not empty
```

Comment:
But the user won't be able to see the errMsg that way. (I think?)
Toasts are a way of flashing an error message to the user.  What about this?

```java
boolean error = false;
        StringBuilder errorMessage = new StringBuilder();
        if (username.length() < minUserLen) {
            error = true;
            errorMessage.append("Username too short");
        }
        if (password.length() < minPassLen) {
            if (error) {
                errorMessage.append(", and ");
            }
            error = true;
            errorMessage.append("Password too short");
        }
        errorMessage.append(".");

        if (error) {
            Toast.makeText(LoginActivity.this, errorMessage.toString(), Toast.LENGTH_LONG);
            return;
        }
```